src/Emitters/CurlEmitter.php

```diff
@@ -81,36 +82,61 @@ public function send($buffer, $curl_send = false) {
         // If the sent buffer contains events...
         if (count($buffer) > 0) {
             if ($type == "POST") {
-                $payload = $this->getPostRequest($buffer);
-                $curl = $this->getCurlRequest($payload, $type);
-                array_push($this->curl_buffer, $curl);
-
-                // If debug is on; store the handle and the payload
-                if ($debug) {
-                    array_push($this->debug_payloads, array("handle" => $curl, "payload" => $payload));
-                }
+                array_push($this->stm_buffer, $buffer);
             }
             else {
                 foreach ($buffer as $event) {
-                    $payload = http_build_query($event);
-                    $curl = $this->getCurlRequest($payload, $type);
-                    array_push($this->curl_buffer, $curl);
+                    array_push($this->stm_buffer, $event);
                 }
             }
         }
 
-        if (count($this->curl_buffer) >= $this->curl_limit) {
-            return $this->rollingCurl($this->curl_buffer, $debug);
+        if (count($this->stm_buffer) >= $this->curl_limit) {
+            return $this->rollingPayload($this->stm_buffer, $type, $debug);
```

Comment:
I'm not sure this had a good name before, what does `rolling` represent in this context? This might be me misunderstanding but from the outside, I'm not sure what I'd expect this function to do.

Reply:
It was a bad name indeed, because it does nothing in a rolling fashion, just that eventually calls `rollingCurl`. I renamed to `mkCurlRequests`, which is closer to what it essentially does.